### ISSUE TITLE
node_modules folder from pnpm install results in "Illegal characters in path." ArgumentException and "**" (double asterisk) folder in folder view

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/aspnetcore#28643]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity2.visualstudio.com/t/node_modules-folder-from-pnpm-install-re/1261486?preview=true)._

---
[severity:It's more difficult to complete my work]
**tl;dr**
When using pnpm inside a folder of a project which contains Razor-Views in Visual Studio, the project-load crashes with the message "illegal characters in path". Workaround is to **exclude the node_modules folder** from the project.

### Prerequisite:
- Use Visual-Studio 2019 version 16.8.1 (either Professional or Enterprise edition)
- create new solution (ASP.NET Core Web Application => ASP.NET Core Web App (Model-View-Controller) using ASP.NET Core 5.0)
- install pnpm (`npm install -g pnpm`)
- install vue cli (`npm install -g @vue/cli`)

### Reproduction steps:
- open powershell in solution-folder
- execute command `vue create frontend`
- pick "Default ([Vue 2] babel, eslint)" and wait for finish (if asked to pick pnpm or npm, choose pnpm)
- in Visual Studio, unload Web-Project and reload project

### Actual behavior:
- Visual Studio fails to load project and only shows "**" folder next to "wwwroot"
- an error pops up "Visual Studio ran into an unexpected problem, and may affet come project system functions." (see attached log-file)
- Building the project in Visual Studio results in MSBuild errors "Resource file "**/*.resx" cannot be found.", "The "GenerateResource" task failed unexpectedly." and "System.ArgumentException: Illegal characters in path."

### Expected behavior:
- Visual Studio load the project normally and shows the frontend folder in the Solution Explorer
- no error message appears and project can be started normally

### Temporary workaround
To avoid this problem, you have to manually exclude the node_modules folder from the project. Either you do this by including the folder in the project and exclude it again to make it disappear or you can add the following ItemGroup to the .csproj file:
><ItemGroup>
    <Compile Remove="frontend\node_modules\**" />
    <Content Remove="frontend\node_modules\**" />
    <EmbeddedResource Remove="frontend\node_modules\**" />
    <None Remove="frontend\node_modules\**" />
</ItemGroup>

(remember to adjust the folder "frontend" to your needs)

---
#### Feedback Bot on 11/19/2020, 05:58 PM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

#### Feedback Bot on 11/19/2020, 08:57 PM: 

<p>Thank you for sharing your feedback! Our teams prioritize action on product issues with broad customer impact. See details at: <a target='_blank' href="https://docs.microsoft.com/en-us/visualstudio/ide/report-a-problem?view=vs-2019#faq">https://docs.microsoft.com/en-us/visualstudio/ide/report-a-problem?view=vs-2019#faq</a>. In case you need answers to common questions or need assisted support, be sure to use <a target='_blank' href="https://visualstudio.microsoft.com/vs/support/">https://visualstudio.microsoft.com/vs/support/</a>. We’ll keep you posted on any updates to this feedback.</p>

#### Lifeng Lu [MSFT] on 12/3/2020, 03:05 PM: 

<p>can you try to add this to the project:</p>
<pre><code>&lt;DefaultExcludesInProjectFolder&gt;$(DefaultExcludesInProjectFolder);frontend\node_modules\**&lt;/DefaultExcludesInProjectFolder&gt;
</code></pre>
<p>similar to the earlier comments, you need replace [frontend] to the real folder the node_modules folder lives. This prevents msbuild to expand globs inside the folder upfront. You can archive it by using  mentioned earlier, but using ‘Remove’, msbuild will expand globs, then remove those items when it reaches the ‘Remove’ item. (Msbuild is a programming language executes in sequence, and anything between the ‘include’ to the ‘remove’ will see the whole set.</p>
<p>Node model produces huge folder which costs lots of time &amp; memory to expand them into msbuild evaluation model (and when it is changed, it triggers project to reevaluate), so it is highly recommended to do, even you don’t run into an error mentioned in this feedback.</p>
<p>Now, go back to the error itself, it is coming from Razor tool, I will loop this issue to them.</p>

---
#### Christoph Keller solved on 11/19/2020, 00:37 AM, 0 votes: 

<p>[Workaround]<br>
To avoid this problem, you have to manually <strong>exclude the node_modules folder from the project</strong>. Either you do this by including the folder in the project and exclude it again to make it disappear or you can add the following ItemGroup to the .csproj file:</p>
<blockquote></blockquote>

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us. We're closing this issue as there was no community involvement here for quite a while now.

You can learn more about our triage process and how we handle issues by reading our [Triage Process](https://github.com/dotnet/aspnetcore/blob/master/docs/TriageProcess.md) writeup.

**Comment by ckeller81** [comment]
Seen that this issue was closed, but I ran again into this today. :)
So in my opinion, this issue should be still active, as this is a razor-tool error as stated by Lifeng Lu [MSFT] in the original issue.

Just to be clear: i am completely ok with the workaround presented in the initial comment, but I think this should be solved differently directly in the razor-tool itself.

@Pilchie is there any more info I could provide, to help finding and fixing this issue?
It seems to be an issue with the sym-linked files created by pnpm, perhaps this helps? :)

Thanks in advance for the help,
Chris
